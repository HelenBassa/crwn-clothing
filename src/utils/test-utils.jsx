import { render } from '@testing-library/react';
import { BrowserRouter as Router } from "react-router-dom";
import { Provider } from "react-redux";
import { store } from "../store/store";


afterAll(() => jest.clearAllMocks());

const AllTheProviders = ({ children }) => {
  return (
    <Provider store={store}>
      <Router>
        {children}
      </Router>
    </Provider>
  );
};

const customRender = (ui, options) => render(ui, { wrapper: AllTheProviders, ...options });

export * from '@testing-library/react';
export * from '@testing-library/jest-dom';
export * from '@testing-library/user-event';

export { customRender as render };