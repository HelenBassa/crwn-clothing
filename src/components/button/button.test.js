import userEvent from "@testing-library/user-event";

import { render, screen } from "../../utils/test-utils";
import Button from "./button.component";

test("should render Button component", () => {
  const onClick = jest.fn();
  const { container } = render(<Button onClick={onClick} />);
  expect(container).toMatchSnapshot();

  userEvent.click(screen.getByRole("button"));
  expect(onClick).toHaveBeenCalled();
});
