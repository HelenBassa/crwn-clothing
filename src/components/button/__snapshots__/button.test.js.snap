// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Button component 1`] = `
<div>
  <button
    class="sc-dkrFOg KWCcW"
  />
</div>
`;
